Automattic/woocommerce-payments
# Remove payment button when product price is 0

Fixes #5406 

#### Changes proposed in this Pull Request


For products with 0 price, on product price, or for 0 price cart, the payment request button ( Google Pay/Apple Pay/Link..) should not be present .






#### Testing instructions





* Go to `Woocommerce -> Settings -> Express checkout` and enable Apple Pay / Google Pay
* Add a product with price 0
* Go to product page. The payment request button shouldn't be shown.
* Add the 0 price product in cart. Check cart and checkout page. Payment request button should not be shown
* Add a shipping fee. Payment request button should appear.

Note: 

If Google Pay is not shown on your environment:
1. Login with your Google account in Chrome browser
2. Go to [https://pay.google.com](https://href.li/?https://pay.google.com/gp/w/u/0/home/paymentmethods), and in the Payment Methods section add a credit card with a US address. At this point, we don’t have a test credit card for Google Pay.
3. We need to access the environment with a secured connection (HTTPS). For this, we can use Jurassic Tube or Ngrok.

If Apple Pay is not shown on your environment, check the instructions here: pdjVC1-11t-p2

-------------------


**Post merge**
